app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -82,18 +82,20 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(feed).try(:properties).try { |prop| prop['Machine Id'] }
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
         os = os_for(feed)
-        os_resources = connection.inventory.list_resources_for_type(os.path, true)
-        unless os_resources.nil? || os_resources.empty?
-          return os_resources.first
+        unless os.nil?
+          os_resources = connection.inventory.list_resources_for_type(os.path, true)
+          unless os_resources.nil? || os_resources.empty?
+            return os_resources.first
+          end
         end
 
-        $mw_log.warn "Found no OS resources for resource type #{os.path}"
+        $mw_log.warn "Found no OS resources for resource type #{os.try(:path)}"
```

Comment:
yeah, it makes sense putting it to that block. I'll do that. But there is no point in printing something like `Found no OS resources for resource type nil` in this place, because the `os_for` method already does that.